### ISSUE TITLE
Bitmap emoji fonts do not render

### ISSUE THREAD

**Issue by nt8r** [zed-industries/zed#15925]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Fonts such as the Samsung Emoji font can be selected as fallback fonts, but emoji render as unfilled rectangles rather than displaying the actual emoji glyph. The font in question is a `.ttf` containing bitmap symbols (`fc-query` states `outline: False(s)` and `scalable: True(s)`). The font *is* accepted in the font settings and listed in the acceptable values for those settings.

If I install a SVG-based emoji font, such as Noto Emoji, then that font will be used regardless of preferences.

### Environment

Zed: v0.149.0 (Zed Dev 6f6eeb65954bb96954522deb330e40b746075ba6)
OS: Linux Wayland arch unknown
Memory: 15.3 GiB
Architecture: x86_64
GPU: Intel(R) Xe Graphics (TGL GT2) || Intel open-source Mesa driver || Mesa 24.1.5-arch1.1

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>

</pre></details>

**Comment by nt8r** [comment]
I don't believe this is the same as #12629 because the font in question is a `.ttf`, but possibly there is some overlap.

**Comment by gbatiz** [comment]
I have tried this with many kinds of fonts, and it is consistently broken. We use emojis in markdown docs quite a lot, and my main use-case for using zed would be note taking and markdown editing.

System: Arch Linux, latest + Zed 0.149.3

See comparison with Sublime Text
![image](https://github.com/user-attachments/assets/e823f8d7-927a-48c3-a6b4-293861a7fc7b)

**Comment by nt8r** [comment]
This bug still reproduces. Stale bots are user-hostile.

**Comment by nt8r** [comment]
@gusbemacbe for what it's worth, COLRv0 is not a bitmap font, but it does seem likely that the font rendering stack used by Zed might be missing support for it. But that's probably a separate problem considering that the settings do allow the selection of bitmap emoji fonts; they just render incorrectly.

**Comment by gusbemacbe** [comment]
Good morning, @nt8r! Shall I create a separated problem issue here and report this problem?

**Comment by gbatiz** [comment]
At this point, I'm just gonna unsubscribe. VSCode with a dedicated text editor profile is more than capable for my use case.

**Comment by nt8r** [comment]
This bug is still reproducible. Stale bots are still user-hostile.

**Comment by dpalvolgyi-pw** [comment]
Checked on Zed 1.0.0 5ec84a926ef83865afb92d2a3d1ca3b419572cf9

Tired with otf font (instead of `.ttf`), the issue remains

**Comment by nt8r** [comment]
> The font _is_ accepted in the font settings and listed in the acceptable values for those settings.

I no longer see any emoji fonts (vector or bitmap) in the list of accepted values for font settings, as of Zed 1.12.1. So this bug is not stale, nor fixed, but seems to be blocked by an additional regression (#61660).
